### PR TITLE
MULE-11022: Http Conficts with Wildcard in the middle of the path.

### PR DESCRIPTION
**Pull request by facundovs**
Fix in HttpListenerRegistry.
Fix in HttpListerRegistryTestCase.
paths-collides.xml configuration file was created.
PathsCollidesTestCase.java

### CODE REVIEW

File: modules/http/src/main/java/org/mule/module/http/internal/listener/HttpListenerRegistry.java [hunk 1]
```diff
@@ -295,13 +313,23 @@ private void addCatchAllPathMapIfNotNull(PathMap currentPathMap, Stack<PathMap>
...
+        private RequestHandlerMatcherPair findRequestHandlerMatcherPair(List<RequestHandlerMatcherPair> requestHandlerMatcherPairs, HttpRequest request, String path)
         {
+            final String[] pathParts = splitPath(path);
             for (RequestHandlerMatcherPair requestHandlerMatcherPair : requestHandlerMatcherPairs)
             {
                 if (requestHandlerMatcherPair.getRequestMatcher().matches(request))
                 {
-                    return requestHandlerMatcherPair;
+                    String requestHandlerPath = requestHandlerMatcherPair.getRequestMatcher().getPath();
+                    if(requestHandlerPath.contains("*") && ! requestHandlerPath.endsWith("*") && ! requestHandlerPath.endsWith("*/") ){
```
Comment (pablokraan) [inline modules/http/src/main/java/org/mule/module/http/internal/listener/HttpListenerRegistry.java]: Add constant for "*"

Comment (pablokraan) [inline modules/http/src/main/java/org/mule/module/http/internal/listener/HttpListenerRegistry.java]: Fix code style

File: modules/http/src/main/java/org/mule/module/http/internal/listener/HttpListenerRegistry.java [hunk 2]
```diff
@@ -295,13 +313,23 @@ private void addCatchAllPathMapIfNotNull(PathMap currentPathMap, Stack<PathMap>
...
+                        String vector [] = requestHandlerMatcherPair.getRequestMatcher().getPath().split("\\*");
+                        for(String pathPart : pathParts){
+                            if(vector.length>0 && ! pathPart.equals("") && vector[1].contains(pathPart))
```
Comment (pablokraan) [inline modules/http/src/main/java/org/mule/module/http/internal/listener/HttpListenerRegistry.java]: pathPart.equals("") does not mean vector.length==0?

File: modules/http/src/test/java/org/mule/module/http/internal/listener/PathsCollidesTestCase.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+    @Rule
+    public SystemProperty systemPropertyResponse1 = new SystemProperty("response1", response1);
+
+    @Rule
+    public SystemProperty systemPropertyResponse2 = new SystemProperty("response2", response2);
+
+    private final String path;
+    private final String response;
+    private final String URL = "http://localhost:%s/%s";
```
Comment (pablokraan) [inline modules/http/src/test/java/org/mule/module/http/internal/listener/PathsCollidesTestCase.java]: Make this static final and move next to the other constants

File: modules/http/src/main/java/org/mule/module/http/internal/listener/HttpListenerRegistry.java (hunk 2 above)
Comment (facundovs) [inline modules/http/src/main/java/org/mule/module/http/internal/listener/HttpListenerRegistry.java]: No. They are differents variables.
Vector contains the differents parts of the path of the requestMatcher.
pathPaths contains the differents parts of the current request path.

**Pull request merged**